<launch>
    <arg name="robot_type" default="$(env ROBOT_TYPE)" doc="Robot type: [ti5]"/>
    

    <param name="legged_robot_description" textfile="$(find legged_ti5_description)/urdf/$(arg robot_type).urdf"/>
    <param name="use_imu_topic" value="true" />

    <param name="leg3_kp_scale" value="1.0"/>
    <param name="leg3_kd_scale" value="1.0"/>
    <param name="leg4_kp_scale" value="0.6"/>
    <param name="leg4_kd_scale" value="0.6"/>
    <param name="leg5_kp_scale" value="1.0"/>
    <param name="leg5_kd_scale" value="1.0"/>
    

    <node name="generate_urdf" pkg="legged_common" type="generate_urdf.sh" output="screen"
          args="$(find legged_ti5_description)/urdf/$(arg robot_type).urdf $(arg robot_type)"/>

    <rosparam file="$(find legged_ti5_hw)/config/$(arg robot_type).yaml" command="load"/>
  
    <node name="legged_ti5_hw" pkg="legged_ti5_hw" type="legged_ti5_hw" respawn="false" clear_params="true" output="screen"/>
    <!-- <node name="hw_lcm_node" pkg="hw_lcm" type="hw_lcm_node" output="screen"/> -->

    <!-- <node pkg="serial_imu" type="serial_imu" name="imu_node" output="screen"/> -->
    <!-- <node pkg="yesense_imu" type="yesense_imu_node" name="yesense_imu_node"  output = "screen" >
        <param name="yesense_port"      type="string" value="/dev/ttyACM0"/>
        <param name="yesense_baudrate"  type="int"    value="460800"/>

  </node> -->

</launch>
